clippy_utils/src/lib.rs

```diff
@@ -2817,9 +2847,13 @@ pub enum ExprUseNode<'tcx> {
     /// The callee of a function call.
     Callee,
     /// Access of a field.
-    FieldAccess(Ident),
+    FieldAccess(HirId, Ident),
     /// Borrow expression.
     AddrOf(ast::BorrowKind, Mutability),
+    /// Assignment.
+    Assign(usize),
+    /// Assignment with an operator.
+    AssignOp(BinOpKind, usize),
```

Comment:
No need to capture this. The lint already checks if the binding in the closure is mutable so `x = something` would fail to compile.

Reply:
> The lint already checks if the binding in the closure is mutable so `x = something` would fail to compile.

Do you mean this line?

```rust
&& let PatKind::Binding(BindingMode(ByRef::No, Mutability::Not), arg_id, _, None) = param.pat.kind
```

This checks the mut pattern, right. So `Assign` is indeed not needed for this lint, but given that it's a function in clippy_utils, I'd like to make it take all cases into account as much as possible. And `AssignOp` is still needed for something like `x += y`.

Reply:
`x += y` also wouldn't compile. Once you fix `use_mutability` there will be no uses of these variants left (`use_node` and `expr_use_ctxt` shouldn't be used there).